simphony/cuds/abc_dataset.py

```diff
@@ -0,0 +1,174 @@
+
+        Returns
+        -------
+        uids : list of uuid.UUID
+            The uids of the added objects.
+
+        Raises
+        ------
+        ValueError :
+            when there is an object with an uids that already exists
+            in the dataset.
+        """
+
+    @abstractmethod
+    def update(self, iterable):  # pragma: no cover
+        """Updates a set of objects from the provided iterable.
+
+        Takes the uids of the objects and searches inside the dataset for
+        those objects. If the object exists, they are replaced in the
+        dataset. If any object doesn't exist, it will raise an exception.
+
+        Parameters
+        ----------
+
+        iterable : iterable of objects
+            the objects that will be replaced.
+
+        Raises
+        ------
+        ValueError :
+            If any object inside the iterable does not exist.
+        """
+
+    @abstractmethod
+    def get(self, uid):  # pragma: no cover
+        """Returns a copy of the object with the 'uid' id.
+
+        Parameters
+        ----------
+
+        uid : uuid.UUID
+            the uid of the object
+
+        Raises
+        ------
+        KeyError :
+            when the object is not in the container.
+
+        Returns
+        -------
+        object :
+            A copy of the internally stored info.
+        """
+
+    @abstractmethod
+    def remove(self, uids):  # pragma: no cover
+        """Remove the object with the provided uids from the container.
+
+        The uids inside the iterable should exists in the container. Otherwise
+        an exception will be raised.
+
+        Parameters
+        ----------
+        uids : iterable of uuid.UUID
+            the uids of the objects to be removed.
+
+        Raises
+        ------
+        KeyError :
+            If any object doesn't exist.
+        """
+
+    @abstractmethod
+    def iter(self, uids=None, item_type=None):  # pragma: no cover
+        """Generator method for iterating over the objects of the container.
+
+        It can receive any kind of sequence of uids to iterate over
+        those concrete objects. If nothing is passed as parameter, it will
+        iterate over all the objects.
+
+        Parameters
+        ----------
+        uids : iterable of uuid.UUID, optional
+            sequence containing the uids of the objects that will be
+            iterated. When the uids are provided, then the objects are
+            returned in the same order the uids are returned by the iterable.
+            If uids is None, then all objects are returned by the iterable
+            and there is no restriction on the order that they are returned.
+
+        Yields
+        ------
+        object : Particle
+            The object item.
+
+        Raises
+        ------
+        KeyError :
+            if any of the ids passed as parameters are not in the dataset.
+        """
+
+    @abstractmethod
+    def has(self, uid):  # pragma: no cover
```

Comment:
However, the same applies to all the other pythonic container methods. Perhaps it's better to leave this the subject of another PR. (And I see we have `has_something` methods there)